### PR TITLE
Graphql schema tidyup

### PR DESCRIPTION
**Pull request by AxonC**
- Utilises input objects
- Adds quick styleguide for the project

### CODE REVIEW

File: graphql/bookings.graphql [hunk 1]
```diff
@@ -5,25 +5,25 @@ extend type Query {
...
+    to: DateTime! @rules(apply: ["after:input.from"]),
+}
 
-    deleteBooking(
-        id: ID!
-    ): Booking @delete
+extend type Mutation @middleware(checks: ["auth"]) {
+    createBooking(input: CreateBookingInput! @spread): Booking! @field(resolver: "App\\Modules\\Bookings\\CreateBookingHandler")
+    updateBooking(input: UpdateBookingInput! @spread): Booking! @field(resolver: "App\\Modules\\Bookings\\UpdateBookingHandler")
+    deleteBooking(id: ID!): Booking @delete
```
Comment (atoff) [inline graphql/bookings.graphql]:
```suggestion
    deleteBooking(id: ID!): Booking! @delete
```

File: graphql/schema.graphql [hunk 2]
```diff
@@ -15,14 +15,25 @@ input DateRange {
...
-    id: ID
+    id: ID!
     atcRating: Rating!
 }
 
 type Rating {
     code: String!
 }
 
+interface Node {
```
Comment (atoff) [inline graphql/schema.graphql:26]: Not quite sure how useful having this as an interface is?

File: graphql/endorsement.graphql [hunk 3]
```diff
@@ -4,30 +4,29 @@ extend type Query {
...
-    ): SpecialEndorsement @update
+    updateSpecialEndorsement(id: ID!, input: SpecialEndorsementInput! @spread): SpecialEndorsement @update
 
-    deleteSpecialEndorsement(
-        id: ID!
-    ): SpecialEndorsement! @delete
+    deleteSpecialEndorsement(id: ID!): SpecialEndorsement! @delete
 
     grantSpecialEndorsement(
         request_id: ID!
```
Comment (atoff) [inline graphql/endorsement.graphql:25]: Use another input type here?

File: graphql/schema.graphql [hunk 4]
```diff
@@ -15,14 +15,25 @@ input DateRange {
...
+    id: ID!
+}
+
+type Query {
+    node(
```
Comment (atoff) [inline graphql/schema.graphql:31]: What is this?

**Review by atoff — CHANGES_REQUESTED** [review]

File: graphql/endorsement.graphql (hunk 3 above)
Comment (AxonC) [reply graphql/endorsement.graphql:25]: There's only one parameter - see the section in the styleguide about this.

File: graphql/schema.graphql (hunk 4 above)
Comment (AxonC) [reply graphql/schema.graphql:31]: https://atheros.ai/blog/graphql-interfaces-and-unions-how-to-design-graphql-schema have a gander at this

File: graphql/schema.graphql (hunk 2 above)
Comment (AxonC) [reply graphql/schema.graphql:26]: https://atheros.ai/blog/graphql-interfaces-and-unions-how-to-design-graphql-schema

File: graphql/endorsement.graphql (hunk 3 above)
Comment (AxonC) [reply graphql/endorsement.graphql:25]: https://github.com/VATSIM-UK/ukts-core-api/pull/38/files#diff-1e4ce90eff22fd15fd458e414f3d8cccR30-R35

Comment (atoff) [reply graphql/endorsement.graphql:25]: Fair enough, just checking for consistency

**Review by atoff — APPROVED** [review]

**Pull request merged**